test/lit.cfg.py

```diff
@@ -100,6 +100,16 @@ def one_line(s):
             %t.interpreter $in -1 /dev/stdout | diff - $out || (echo $in && exit 1)
         done
     ''')),
+    
+    ('%check-search-convert-test', one_line('''
+        for out in %test-dir-out/*.out.diff; do
+            in=%test-dir-in/`basename $out .out.diff`.in
+            %t.interpreter $in 3 /dev/stdout | diff - $out
+            %kore-convert $in -o %t.bin && %t.interpreter %t.bin 3 /dev/stdout | diff - $out
+            %t.interpreter $in 3 %t.bin --binary-output
+            %kore-convert %t.bin -o %t.kore && %kore-convert $out --to=text | diff - %t.kore
+        done
+    ''')),
```

Comment:
Yes, but this particular substitution hard-codes precisely 3 rewrite steps into the test. I would prefer to keep the substitutions in the test suite _general_ (and have unusual test cases, that for example don't rewrite fully, have longer test headers), rather than only being applicable to the single test case that you use it in.

Reply:
Ok!